BookService/Logic/LibraryCollection.cs

```diff
@@ -0,0 +1,318 @@
+        /// or book's index out of range of their collection's count</exception>
+        public void UpdateAuthor(int authorId, int bookId)
+        {
+            if (bookAuthors.FindAll(item => item.AuthorIndex == authorId
+                                             && item.BookIndex == bookId).Count == 0)
+            {
+                bookAuthors.Add(new BookAuthor()
+                {
+                    BookIndex = bookId, AuthorIndex = authorId
+                });
+            }
+        }
+
+        /// <summary>
+        /// Add author to collection
+        /// </summary>
+        /// <param name="author">New author</param>
+        public void AddAuthor(Author author)
+        {
+            authors.Add(author);
+        }
+
+        /// <summary>
+        /// Search books by it's genre
+        /// </summary>
+        /// <param name="genreIndex">Index of the genre</param>
+        /// <returns>Collection of books</returns>
+        public IEnumerable<Book> SearchByGenre(int genreIndex)
+        {
+            var gettedBooks = from item in bookGenres
+                              where item.GenreIndex == genreIndex
+                              select books[item.BookIndex];
+
+            return gettedBooks;
+        }
+
+        /// <summary>
+        /// Search books by it's author
+        /// </summary>
+        /// <param name="authorIndex">Index of the genre</param>
+        /// <returns>Collection of books</returns>
+        public IEnumerable<Book> SearchByAuthor(int authorIndex)
+        {
+            var gettedBooks = from item in bookAuthors
+                              where item.AuthorIndex == authorIndex
+                              select books[item.BookIndex];
+
+            return gettedBooks;
+        }
+
+        /// <summary>
+        /// Get genre value by it's index
+        /// </summary>
+        /// <param name="id">Index of the selected genre</param>
+        /// <returns>genre by selected index</returns>
+        /// <exception cref="ArgumentOutOfRangeException">Throw when index out of
+        /// list's count range</exception>
+        public Genre GetGenreByIndex(int id)
+        {
+            return genres[id];
+        }
+
+        /// <summary>
+        /// Return collection of genres
+        /// </summary>
+        /// <returns>IEnumerable collection</returns>
+        public IEnumerable<Genre> GetGenres()
+        {
+            return genres;
+        }
+
+        /// <summary>
+        /// Remove genre by it's index
+        /// </summary>
+        /// <param name="id">Index of the selected genre</param>
+        /// <returns>Deleted Genre</returns>
+        /// <exception cref="ArgumentOutOfRangeException">Throw if index out of range</exception>
+        public Genre RemoveGenre(int id)
+        {
+            Genre deletedGenre = null;
+            if (bookGenres.FindAll(items => items.GenreIndex == id).Count == 0)
```

Comment:
if you need count y can use    bookGenres. Count(items => items.GenreIndex == id) == 0